### PR TITLE
fix concurrency issue when upgrade

### LINKED ISSUE #1145: [BUG] 延迟删除的mount pod平滑升级后立刻被删除
**What happened**:
1、配置mount pod   juicefs/mount-delete-delay: 3h ,创建应用pod
2、等应用pod  completed后进行平滑升级
3、mount pod 被删除，预期不应该被删除
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS CSI Driver version (which image tag did your CSI Driver use):
- Kubernetes version (e.g. `kubectl version`):
- Object storage (cloud provider and region):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

### PR DESCRIPTION
**Pull request by zwwhdls**
1. select no complete mount pod when mount/umount
2. remove finalizer until new mount pod created successfully when old is deleted
3. send SIGHUP only once when upgrade
4. check old mount pod support smooth upgrade 
5. fix https://github.com/juicedata/juicefs-csi-driver/issues/1145

### CODE REVIEW

File: pkg/controller/pod_driver.go [hunk 1]
```diff
@@ -438,49 +450,42 @@ func (p *PodDriver) podDeletedHandler(ctx context.Context, pod *corev1.Pod) (Res
...
-			newPod, err := p.newMountPod(ctx, pod, newPodName)
-			if err == nil {
-				_, err = p.Client.CreatePod(ctx, newPod)
-				if err != nil {
-					log.Error(err, "Create pod")
-				}
+	if notAvailable {
+		// create pod
+		newPodName := podmount.GenPodNameByUniqueId(pod.Labels[common.PodUniqueIdLabelKey], true)
+		log.Info("pod targetPath not empty, need to create a new one", "newPodName", newPodName)
```
Comment (zxh326) [inline pkg/controller/pod_driver.go]: why not available mount pods mean **pod targetPath not empty**?

File: pkg/controller/pod_driver.go [hunk 2]
```diff
@@ -968,13 +973,13 @@ func (p *PodDriver) needCreateMountPod(ctx context.Context, uniqueId, hashVal st
...
-	needCreate := true
+	notAvailable := true
 	for _, po := range pods {
 		if po.DeletionTimestamp == nil && !resource.IsPodComplete(&po) {
-			needCreate = false
+			notAvailable = false
 		}
 	}
-	return needCreate, nil
+	return notAvailable, nil
```
Comment (zxh326) [inline pkg/controller/pod_driver.go]: func name is has available mount, but return not available pod, It's very confusing.

File: pkg/controller/pod_driver.go (hunk 1 above)
Comment (zxh326) [reply pkg/controller/pod_driver.go]: How to manually delete a pod in a delayed deletion state(with no refs)

Comment (zwwhdls) [reply pkg/controller/pod_driver.go]: not available means there are no available mount pod in cluster. If pod targetPath not empty and no available , need to create a new one.

Comment (zwwhdls) [reply pkg/controller/pod_driver.go]: If manually delete the delay deleted pod, it will not be handled as we discussed before.

Comment (zxh326) [reply pkg/controller/pod_driver.go]: If I manually delete a delay pod, will a new one be created? even if there's no reference?

Comment (zxh326) [reply pkg/controller/pod_driver.go]: > not available means there are no available mount pod in cluster. If pod targetPath not empty and no available , need to create a new one.

should we check `len(existTargets) > 0` here?

check `len(existTargets) == 0` before but with other condition..

https://github.com/juicedata/juicefs-csi-driver/blob/895ddad32d50e0325fb469aec1a6f97a12ed17e8/pkg/controller/pod_driver.go#L435

**Review by zxh326 — APPROVED** [review]

**Pull request merged**